conan/conanfile.py

```diff
@@ -0,0 +1,163 @@
+        "with_jemalloc": True,
+        "with_mongodb": False, 
+        "with_postgresql": False,
+        "with_redis": False,
+        "with_grpc": False, 
+        "with_clickhouse": False,
+        "with_universal": False,
+        "namespace": "userver",
+        "namespace_begin": "",
+        "namespace_end": ""
+    }
+
+    scm = {
+        "type": "git",
+        "url": "https://github.com/jorgenpo/userver.git",
+        "revision": "conan"
+    }
+
+    _build_subfolder = "cmake-build"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if not self.options.namespace_begin:
+            self.options.namespace_begin = f"namespace {self.options.namespace} {{"
+        if not self.options.namespace_end:
+            self.options.namespace_end = "}"
+
+    def requirements(self):
+        self.requires("boost/1.79.0")
+        self.requires("libev/4.33")
+        self.requires("spdlog/1.9.0")
+        self.requires("fmt/8.1.1")
+        self.requires("c-ares/1.18.1")
+        self.requires("libcurl/7.68.0")
+        self.requires("cryptopp/8.6.0")
+        self.requires("yaml-cpp/0.7.0")
+        self.requires("cctz/2.3")
+        self.requires("http_parser/2.9.4")
+        self.requires("openssl/1.1.1q")
+
+        if self.options.with_jemalloc:
+            self.requires("jemalloc/5.2.1")
+        if self.options.with_grpc:
+            self.requires("grpc/1.48.0")
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["CMAKE_FIND_DEBUG_MODE"] = "OFF"
+
+        cmake.definitions["USERVER_OPEN_SOURCE_BUILD"] = "ON"
+        cmake.definitions["USERVER_IS_THE_ROOT_PROJECT"] = "OFF"
+        cmake.definitions["USERVER_DOWNLOAD_PACKAGES"] = "OFF"
+        cmake.definitions["USERVER_NAMESPACE"] = self.options.namespace
+        cmake.definitions["USERVER_NAMESPACE_BEGIN"] = self.options.namespace_begin
+        cmake.definitions["USERVER_NAMESPACE_END"] = self.options.namespace_end
+
+        if not self.options.with_jemalloc:
+            cmake.definitions["USERVER_FEATURE_JEMALLOC"] = "OFF"
+        if not self.options.with_mongodb:
+            cmake.definitions["USERVER_FEATURE_MONGODB"] = "OFF"
+        if not self.options.with_postgresql:
+            cmake.definitions["USERVER_FEATURE_POSTGRESQL"] = "OFF"
+        if not self.options.with_redis:
+            cmake.definitions["USERVER_FEATURE_REDIS"] = "OFF"
+        if not self.options.with_grpc:
+            cmake.definitions["USERVER_FEATURE_GRPC"] = "OFF"
+        if not self.options.with_clickhouse:
+            cmake.definitions["USERVER_FEATURE_CLICKHOUSE"] = "OFF"
+        if not self.options.with_universal:
+            cmake.definitions["USERVER_FEATURE_UNIVERSAL"] = "OFF"
+
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        # Rename some packages for cmake to find them in find_package
```

Comment:
Please add tests for conan packaging into Github CI.

Some day, we may change the package names we use. CI would help us to avoid breakage of conan scripts

Reply:
Looks like it should be a patch (for conan) not a direct statements in the conanfile. Because package names in userver differ conan package names and if userver changes package names in one release, you cannot add new release without changing conanfile. If you use this as a patch, you may change only a patch for specific version not recipe.